lib/cucumber/formatter/html.rb

```diff
@@ -25,88 +26,56 @@ def initialize(runtime, path_or_io, options)
         @runtime = runtime
         @options = options
         @buffer = {}
-        @builder = create_builder(@io)
+        @builder = Cucumber::HtmlBuilder.new(target: @io, indent: 0)
         @feature_number = 0
         @scenario_number = 0
         @step_number = 0
         @header_red = nil
         @delayed_messages = []
         @img_id = 0
         @text_id = 0
-        @inside_outline = false
+        @inside_outline        = false
         @previous_step_keyword = nil
       end
 
       def embed(src, mime_type, label)
-        case(mime_type)
-        when /^image\/(png|gif|jpg|jpeg)/
-          unless File.file?(src) or src =~ /^data:image\/(png|gif|jpg|jpeg);base64,/
-            type = mime_type =~ /;base[0-9]+$/ ? mime_type : mime_type + ';base64'
-            src = 'data:' + type + ',' + src
-          end
-          embed_image(src, label)
-        when /^text\/plain/
-          embed_text(src, label)
+        if image?(mime_type)
+          src = src_is_file_or_data?(src) ? src : "data:#{standardize_mime_type(mime_type)},#{src}"
+
+          @builder.embed_image(src: set_path(src), label: label, id: "img_#{@img_id}")
```

Comment:
I'd use a private `attr_reader` for `builder` here. Sandi Metz taught me to basically never use instance variables.